lib/src/fake_converted_query.dart

```diff
@@ -32,6 +32,9 @@ class FakeConvertedQuery<T extends Object?> extends FakeQueryWithParent<T> {
   @override
   FakeQueryWithParent? get parentQuery => _nonConvertedParentQuery;
 
+  @override
+  Query<T> limit(int limit) => this;
```

Comment:
After looking more closely to the current implementation, I realize that `MockQuery` already implements `limits` and does so correctly: it actually clips results.
https://github.com/atn832/fake_cloud_firestore/blob/c71a47625b5f819fce8fab9f30b2b8a50e23ff02/lib/src/mock_query.dart#L195

We even have a unit test for it:
https://github.com/atn832/fake_cloud_firestore/blob/c71a47625b5f819fce8fab9f30b2b8a50e23ff02/test/mock_query_test.dart#L173

Since `MockQuery` and `FakeConvertedQuery` both extend `FakeQueryWithParent`, the correct way to fix `limit` for converted queries is to move `MockQuery`'s `limit` implementation to `FakeQueryWithParent`. Then `limit` should work for both queries with and without converters.

Would you mind removing this code and moving `MockQuery`'s `limit` implementation to `FakeQueryWithParent`?

Reply:
Doing so results in the following error - could you please advise?

```
Type: FakeQueryWithParent<T>

The argument type 'FakeQueryWithParent<T>' can't be assigned to the parameter type 'MockQuery<T>?'.
(https://dart.dev/diagnostics/argument_type_not_assignable)

Reply:
You are right. I had forgotten how queries were implemented and what I suggested earlier about moving the code to `FakeQueryWithParent` does not work. I think we should fall back to implementing a custom implementation in `FakeConvertedQuery` like you first did. I tried this, and it seems to work. Would you mind trying this:
```dart
// in fake_converted_query.dart
  @override
  Query<T> limit(int limit) {
    return FakeConvertedQuery<T>(
        _nonConvertedParentQuery.limit(limit) as FakeQueryWithParent,
        _converter);
  }
```
What this does is, if someone calls `convertedQuery.limit(k)`, it will return a converted query whose parent query is the nonConvertedQuery to which limit is applied. This way, we reuse MockQuery's implementation. The unit test passes. I had to tweak the unit test as I found out what I wrote earlier does not work (see comment in the other thread).

In the long term, we should delete `FakeConvertedQuery` and make `MockQuery` support the converted case, just like I did for `MockCollectionReference`. I just filed #236 to keep track of it. This will enable us to support all the other methods to the converted case (startAt, endAt, etc).